types/wordpress__blocks/api/registration.d.ts

```diff
@@ -167,3 +176,29 @@ export function unregisterBlockStyle(blockName: string, styleVariationName: stri
  * @returns The previous block value if successfully unregistered, otherwise `undefined`.
  */
 export function unregisterBlockType(name: string): Block<any> | undefined;
+
+/**
+ * Returns an array with the variations of a given block type.
+ *
+ * @param blockName - Name of block (example: “core/columns”).
+ * @param [scope] - Block variation scope name.
+ *
+ * @returns Block variations.
+ */
+export function getBlockVariations(blockName: string, scope?: BlockVariationScope): BlockVariation[] | undefined;
+
+/**
+ * Registers a new block variation for the given block type.
+ *
+ * @param blockName - Name of the block (example: “core/columns”).
+ * @param variation - Object describing a block variation.
+ */
+export function registerBlockVariation(blockName: string, variation: BlockVariation): void;
```

Comment:
While this function is free to impose its own limits it's worth noting that `addBlockVariations`, which it calls, accepts an array of variations, and in practice I assume this happens because nothing prohibits it.

we may want to consider adding that here and proposing a fix in Gutenberg to update its existing JSDoc type to match.

```ts
export function registerBlockVariation(blockName: string, variation: BlockVariation|BlockVariation[]): void
```

Reply:
Good catch on both I'll update it here and follow up with a PR on Gutenberg's side.